### PR TITLE
hotfix/CP-6178-flutter-show-topics-dialog-not-working

### PR DESCRIPTION
**Pull request by unnaticleverpush**
topic dialog is not visible because of the theme.

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 1]
```diff
@@ -2606,15 +2611,25 @@ public void showTopicsDialog(Activity dialogActivity, TopicsDialogListener topic
...
-        });
-        alert.show();
+        try {
+          AlertDialog.Builder alertBuilder =
+                  getTopicAlertBuilder(dialogActivity, topicsDialogListener, themeResId, channelConfig, channelTopics,
+                          hasDeSelectAllInitial, checkboxLayout, unsubscribeCheckbox, selectedTopics);
+          AlertDialog alert = alertBuilder.create();
+          alert.setOnShowListener(dialog -> {
+            Logger.d(LOG_TAG, "showTopicsDialog activity: " + dialogActivity.getClass().getCanonicalName());
+            showingTopicsDialog = true;
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2621]: wont this stop the recursive call from showing the dialog? with L2562 isShowingTopicsDialog

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 2]
```diff
@@ -2606,15 +2611,25 @@ public void showTopicsDialog(Activity dialogActivity, TopicsDialogListener topic
...
+          });
+          Logger.d("Topic Dialog", "Test 1");
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2623]: remove

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 3]
```diff
@@ -2606,15 +2611,25 @@ public void showTopicsDialog(Activity dialogActivity, TopicsDialogListener topic
...
+          if (alert != null) {
+            alert.show();
+          }
+        } catch (Exception e) {
+          Logger.d(LOG_TAG, e.getLocalizedMessage());
+          if (!isRecursiveCall) {
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2629]: do we need to do the whole try catch if isRecursiveCall? i dont think so right? so this if can be around it

**Review by cpKevinrodriguesnobre — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 3 above)
Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:2629]: @cpKevinrodriguesnobre Sorry. I didn't get it. Can you explain it more?

Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:2629]: image
does this help?

Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:2629]: No. It won't work. When it goes to catch and showTopicsDialog at that time isRecursiveCall flag will be true so it won't go inside that if and the dialog will not gonna be open. 

We do need not to call again showTopicsDialog method once it's true. that's why put it on catch block.

**Review by cpKevinrodriguesnobre — APPROVED** [review]

**Pull request merged**